motivaa-backend/motivaa-app/src/test/java/backendtesting/testcases/ExcelOnModulesTest.java

```diff
@@ -1,74 +1,363 @@
+        final String excelInput2Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule2Link.ExcelsOnModulesTestModule2Link::ExcelsOnModulesTestModule2.ExcelsOnModulesTestModule2Link::ExcelAttributeOnParent_InputsOnMultipleModules.ExcelsOnModulesTestModule2Link::input2_shouldNotTriggerExcelUntilAll4InputsHaveValue_notOptional";
+
+        final String input1to3Value = "noValueShouldTriggerTheExcel";
+        final String excelOutput1Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule4Link.ExcelsOnModulesTestModule4Link::ExcelsOnModulesTestModule4.ExcelsOnModulesTestModule4Link::ExcelAttributeOnParent_InputsOnMultipleModules.ExcelsOnModulesTestModule4Link::output1_shouldBeSameAsInput1";
+        final String excelOutput2Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule3Link.ExcelsOnModulesTestModule3Link::ExcelsOnModulesTestModule3.ExcelsOnModulesTestModule3Link::ExcelAttributeOnParent_InputsOnMultipleModules.ExcelsOnModulesTestModule3Link::output2_shouldBeSameAsInput2";
+        final String excelOutput3Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule2Link.ExcelsOnModulesTestModule2Link::ExcelsOnModulesTestModule2.ExcelsOnModulesTestModule2Link::ExcelAttributeOnParent_InputsOnMultipleModules.ExcelsOnModulesTestModule2Link::output3_shouldBeSameAsInput3";
+        final String excelOutput4Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule1Link.ExcelsOnModulesTestModule1Link::ExcelsOnModulesTestModule1.ExcelsOnModulesTestModule1Link::ExcelAttributeOnParent_InputsOnMultipleModules.ExcelsOnModulesTestModule1Link::output4_shouldBeSameAsInput4";
+        final String expectedValueAfterFirstThreeUpdate = "";
+
+        ProcessObject processObject = RequestMaker.createProcessObject(
+                environmentHost,
+                processStartEndpoint,
+                jSessionId);
 
-        String assertionErrorMessage = String.format("When %s gets a value, %s should copy the value to %s", excelInputAttribute, testedExcel, excelOutputAttribute);
+        List<OutputVerification> outputVerificationsForFirstThreeUpdate = Arrays.asList(
+                new OutputVerification(excelOutput1Attribute, expectedValueAfterFirstThreeUpdate),
+                new OutputVerification(excelOutput2Attribute,expectedValueAfterFirstThreeUpdate),
+                new OutputVerification(excelOutput3Attribute,expectedValueAfterFirstThreeUpdate),
+                new OutputVerification(excelOutput4Attribute,expectedValueAfterFirstThreeUpdate)
+        );
 
-        TestUtils.assertJsonPathValue(
-                responseFromFirstUpdate,
-                excelOutputAttribute,
-                testedValue,
-                assertionErrorMessage);
+        TestUtils.updateProcessObjectAndVerifyOutputValues(
```

Comment:
I think here again, I feel like you should only assert the update which is being under test.